server/handlers/validation.go

```diff
@@ -43,8 +43,13 @@ func validateUpdate(cs signed.CryptoService, gun string, updates []storage.MetaU
 	}
 
 	if rootUpdate, ok := roles[data.CanonicalRootRole]; ok {
+		currentRootVersion := builder.GetLoadedVersion(data.CanonicalRootRole)
+		if rootUpdate.Version != currentRootVersion && rootUpdate.Version != currentRootVersion+1 {
```

Comment:
`GetLoadedVersion` returns `1` if there is no root role yet.  Do we want to skip this check if there is no root on the server?  (For instance, if the user is pushing the latest metadata, but not the full history, from one notary server to another).

Non-blocking question:  I was wondering if there was an issue with allowing gaps in versions (handling the 404's on the client side?)  I'm totally ok with enforcing monotonically increasing versions for now, and then allowing for gaps in versions at a later point if we wanted, I am just checking to see if I'm missing an edge case or some other difficulty.